File: pkg/units/unit.go
package units

type Unit struct {
	charterNumber UnitCharterNumber
	kind          UnitKind
	category      UnitCategory
	name          string
	address       string
	city          string
}

func NewUnit(
	charterNumber UnitCharterNumber,
	kind UnitKind,
	category UnitCategory,
	name string,
	address string,
	city string,
) Unit {
	return Unit{
		charterNumber: charterNumber,
		kind:          kind,
		category:      category,
		name:          name,
		address:       address,
		city:          city,
	}
}

func (u Unit) CharterNumber() UnitCharterNumber {
	return u.charterNumber
}

func (u Unit) Kind() UnitKind {
	return u.kind
}

func (u Unit) Category() UnitCategory {
	return u.category
}

func (u Unit) Name() string {
	return u.name
}

func (u Unit) Address() string {
	return u.address
}

func (u Unit) City() string {
	return u.city
}

File: web/capa5/types/unit.ts
export enum UnitKind {
    Group = "group",
    Squadron = "squadron",
    Flight = "flight",
}

export enum UnitCategory {
    Admin = "admin",
    Composite = "composite",
    Cadet = "cadet",
    Senior  = "senior",
}

export class Unit {
    charterNumber: string;
    kind: UnitKind;
    category: UnitCategory;
    name: string;
    address: string;
    city: string;

    constructor(
        charterNumber: string,
        kind: string,
        category: string,
        name: string,
        address: string,
        city: string,
    ) {
        this.charterNumber = charterNumber;
        this.kind = kind
        this.category = category
        this.name = name
        this.address = address
        this.city = city
    }
}